Mage.Sets/src/mage/cards/r/RiaIvorBaneOfBladehold.java

```diff
@@ -0,0 +1,89 @@
+package mage.cards.r;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.BeginningOfCombatTriggeredAbility;
+import mage.abilities.effects.PreventionEffectData;
+import mage.abilities.effects.PreventionEffectImpl;
+import mage.constants.*;
+import mage.abilities.keyword.BattleCryAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.game.Game;
+import mage.game.events.CreateTokenEvent;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.game.permanent.token.PhyrexianMiteToken;
+import mage.game.permanent.token.Token;
+import mage.target.common.TargetCreaturePermanent;
+
+/**
+ *
+ * @author huangn
+ */
+public final class RiaIvorBaneOfBladehold extends CardImpl {
+
+    public RiaIvorBaneOfBladehold(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{W}{B}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.PHYREXIAN);
+        this.subtype.add(SubType.KNIGHT);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(4);
+
+        // Battle cry
+        this.addAbility(new BattleCryAbility());
+
+        // At the beginning of combat on your turn, the next time target creature would deal combat damage to one or more players this combat, prevent that damage. If damage is prevented this way, create that many 1/1 colorless Phyrexian Mite artifact creature tokens with toxic 1 and "This creature can't block."
+        Ability ability = new BeginningOfCombatTriggeredAbility(new RiaIvorBaneOfBladeholdEffect(), TargetController.YOU, false);
+        ability.addTarget(new TargetCreaturePermanent());
+        this.addAbility(ability);
+    }
+
+    private RiaIvorBaneOfBladehold(final RiaIvorBaneOfBladehold card) {
+        super(card);
+    }
+
+    @Override
+    public RiaIvorBaneOfBladehold copy() {
+        return new RiaIvorBaneOfBladehold(this);
+    }
+}
+
+class RiaIvorBaneOfBladeholdEffect extends PreventionEffectImpl {
+
+    public RiaIvorBaneOfBladeholdEffect() {
+        super(Duration.EndOfCombat, Integer.MAX_VALUE, false, false);
+        this.staticText = "the next time target creature would deal combat damage to one or more players this combat, prevent that damage. If damage is prevented this way, create that many 1/1 colorless Phyrexian Mite artifact creature tokens with toxic 1 and \"This creature can't block.\"";
+    }
+
+    public RiaIvorBaneOfBladeholdEffect(final RiaIvorBaneOfBladeholdEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean replaceEvent(GameEvent event, Ability source, Game game) {
+        PreventionEffectData preventionData = preventDamageAction(event, source, game);
```

Comment:
you should check that damage was prevented before continuing after this since the damage may not have been prevented at all

Reply:
check preventionData.getPreventedDamage() > 0.
Question: In case of combat damage deal to more than one player,  whether this effect is triggered in separate instances or not. In another word, is it correct to set the attribute **used** and call discard in replaceEvent().

Reply:
it's kind of an unusual situation, the only situation where this would deal combat damage to multiple players is in the case of an effect redirecting some but not all of the combat damage it deals